lib/src/client/common.dart

```diff
@@ -75,7 +116,7 @@ class ResponseStream<R> extends DelegatingStream<R>
 }
 
 abstract class _ResponseMixin<Q, R> implements Response {
-  ClientCall<Q, R> get _call;
+  ClientCall get _call;
```

Comment:
We should not loose type safety.

Reply:
`.then(...)` allows to transform a `ResponseFuture<R>` to a `ResponseFuture<S>` which will lose type safety. Do you have any suggestion?

```
  @override
  ResponseFuture<S> then<S>(FutureOr<S> Function(R p1) onValue, {Function onError});
```

Reply:
I realized that we don't actually use the type of `call` for anything. So I suggest the following change: it is indeed fine to drop types here, but the we should add types in default constructor to keep initial type safety: 

```
  ResponseFuture(ClientCall<dynamic, R> call) : _call = call, super(call.response
            .fold<R?>(null, _ensureOnlyOneResponse)
            .then(_ensureOneResponse));

```